src/main/java/com/hotels/plunger/Plunger.java

```diff
@@ -58,12 +60,25 @@ public Bucket newBucket(Fields fields, Pipe tail) {
 
   /** Supplies a {@link TapDataWriter} for writing the provided {@link Data} to a {@link Tap} instance. */
   public static TapDataWriter writeData(Data data) {
-    return new TapDataWriter(data);
+    return writeData(data, null);
+  }
+
+  /**
+   * Supplies a {@link TapDataWriter} for writing the provided {@link Data} to a {@link Tap} instance using the given
+   * {@link Configuration} properties.
```

Comment:
I think you can remove the word "properties" here and line 79 and just leave it at Configuration. "properties" is a bit of an overloaded term in Java and might be a bit confusing.